File: frontend/src/api/stats.ts
import { api } from './client'

export interface PlatformCount {
  platform: string
  count: number
}

export interface RecentJob {
  id: string
  platform: string
  status: string
  profiles_collected: number
  created_at: string
}

export interface Stats {
  total_profiles: number
  total_posts: number
  total_clusters: number
  active_jobs: number
  profiles_by_platform: PlatformCount[]
  top_clusters: { id: number; label: string | null; member_count: number }[]
  recent_jobs: RecentJob[]
}

export const statsApi = {
  get: () => api.get<Stats>('/stats'),
}

File: backend/app/schemas/stats.py
from datetime import datetime

from pydantic import BaseModel


class PlatformCount(BaseModel):
    platform: str
    count: int


class RecentJobOut(BaseModel):
    id: str
    platform: str
    status: str
    profiles_collected: int
    created_at: datetime


class StatsOut(BaseModel):
    total_profiles: int
    total_posts: int
    total_clusters: int
    active_jobs: int
    profiles_by_platform: list[PlatformCount]
    top_clusters: list[dict]       # id, label, member_count
    recent_jobs: list[RecentJobOut]
